lib/reversetunnel/srv_test.go

```diff
@@ -38,7 +39,7 @@ func TestServerKeyAuth(t *testing.T) {
 	require.NoError(t, err)
 
 	s := &server{
-		Entry: testlog.FailureOnly(t),
+		log: utils.NewLoggerForTests(logrus.New()),
```

Comment:
`testlog.FailureOnly` has a different behavior - it buffers all logs and only prints them when `t` fails.
please revert this change

Reply:
I believe testing has this behavior by default - it buffers all output (sent via `t.Log(f)` or  `os.StdXXX`) and only outputs it if the test fails regardless of the `-v` (verbose) option. The verbose option is only relevant if one wants to see logs regardless of the outcome.
Do you mean that this should suppress log output regardless of verbosity and only output it on failure?

I initially implemented `InitLoggerForTests` to accept the value of `*testing.T` and route the logs using its `Log(f)` APIs but it became a bit of concern to use with a global logrus logger since it should not outlive a test so I abandoned this approach in favor of writing to `os.Stderr` as this is also buffered by the testing package with the behavior described above.

Reply:
the default behavior in `testing` doesn't know which test outputs which logs.
so if you have a package with 100 tests and 1 of them fails, you get logs for all 100 tests intermixed with test result output.

and yes, the reason I wrote `testlog.FailureOnly` is because it's scoped to a single `testing.T` and not globally

Reply:
The whole point of changes in this PR is exactly this behavior - hence I made this change too. Like you said, it is not going to be helpful if all of the tests' logs are output upon failure (which will happen if this is merged), so I might need to reconsider the approach I've taken with this.

The goal is getting the logs only for the failed test - no doubt about that but also that it should be universal - i.e. all tests should use the FailureOnly logger otherwise it's a hit or miss if used selectively. I'll think about a way to get this goal instead.

Reply:
One other option is to set `logrus.Logger.Out` to a wrapper around `t.Log/c.Log`
It works for both `gocheck` and `testing` tests.

The challenge is making sure that none of our code uses the global logrus logger, but that can be done incrementally afterwards.